### PR TITLE
AzureML Tests Cleanup

### PR DESCRIPTION
**Pull request by pradnyeshjoshi**
### Description

1. Moved AzureML test clusters to the gaming subscription. (Currently CPU, Spark tests are executed on D13v2 machines, and GPU tests are on NC6 machines in the gaming subscription. Will make the GPU cluster NC6sv2 once additional quota is approved - no code change needed then.)
2. Renamed jobs in AzureML test workflows for better readability.
3. Created status badges for `azureml-nightly` workflow in README.
4. Removed unused `wheelfile` argument from `submit_groupwise_azureml_pytest.py`

### Checklist:

- [x] I have followed the [contribution guidelines](../CONTRIBUTING.md) and code style for this project.
- [ ] I have added tests covering my contributions.
- [x] I have updated the documentation accordingly.
- [x] This PR is being made to `staging branch` and not to `main branch`.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -184,6 +184,14 @@ The following tests run on a Linux DSVM daily.
...
 -->
 
+### AzureML Nightly Build Status
+
+Smoke and integration tests are run daily on AzureML.
+
+| Build Type | Branch | Status |  | Branch | Status |
+| --- | --- | --- | --- | --- | --- |
+| **Nightly** | main | [![azureml-nightly](https://github.com/microsoft/recommenders/actions/workflows/azureml-nightly.yml/badge.svg?branch=main)](https://github.com/microsoft/recommenders/actions/workflows/azureml-nightly.yml?query=branch%3Amain) | | staging | [![azureml-nightly](https://github.com/microsoft/recommenders/actions/workflows/azureml-nightly.yml/badge.svg?branch=staging)](https://github.com/microsoft/recommenders/actions/workflows/azureml-nightly.yml?query=branch%3Astaging) |
```
Comment (miguelgfierro) [inline README.md:193]: is it possible to get 3 badges, CPU, GPU and Spark?

**Review by miguelgfierro — APPROVED** [review]
looks good, I just have a minor question

File: README.md (hunk 1 above)
Comment (pradnyeshjoshi) [reply README.md:193]: GitHub badges are workflow specific, not job specific. We could create separate workflows for CPU, GPU, Spark.
PS There is a somewhat convoluted way to get a job status in GitLab as mentioned here https://stackoverflow.com/a/47343010. In our case, for example, CPU status depends on multiple jobs (`group_cpu_001`, `group_cpu_002`, ...) within the workflow. If one of the groups fail, the badge will not be created.

Comment (miguelgfierro) [reply README.md:193]: > If one of the groups fail, the badge will not be created.

when you say that is not created, what exactly happens? do you get a 404 or do you get a red badge? What we want is that if one of the groups fail, we get a red badge, if we click on the badge, then it takes you to the error. Do you think this is possible?

Comment (pradnyeshjoshi) [reply README.md:193]: Let's say we have the CPU badge generation step in the workflow that depends on three jobs, one for each group. If one of the jobs fails, the badge generation step will be cancelled and we will get 404.
I was not able to find a way to link the badge to the latest run/error. The best I could do here is link the badge to the Actions page with the corresponding branch filter applied.

Comment (miguelgfierro) [reply README.md:193]: got it, feel free to merge whenever you want

**Pull request merged**